src/util/network.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/network.h>
+#include <util/strencodings.h>
+#include <string>
+
+SocketAddrError hasValidHostPort(std::string_view socketAddr)
+{
+    uint16_t port{0};
+    std::string hostname;
+    bool validPort = SplitHostPort(socketAddr, port, hostname);
+
+    SocketAddrError res = SocketAddrError::OK;
+    if (!validPort || port == 0)
+        res = SocketAddrError::NO_PORT;
+    if (hostname.empty() || hostname.find(' ') != std::string::npos)
+        res = (res == SocketAddrError::NO_PORT) ? SocketAddrError::NO_HOSTPORT 
```

Comment:
```suggestion
        res = (res == SocketAddrError::NO_PORT) ? SocketAddrError::NO_HOSTPORT
```

to appease the whitespace linter
```
+        res = (res == SocketAddrError::NO_PORT) ? SocketAddrError::NO_HOSTPORT 
^---- failure generated from lint-whitespace.py
```